### ISSUE TITLE
Repeating a successful command after revert + reload fails with caching enabled

### ISSUE THREAD

**Issue by nad** [agda/agda#2959]
If caching is enabled, then one can sometimes get into a situation where a case-split first works and then, after the file has been reverted and reloaded, does not work.

The following files demonstrate the problem:
* `M.agda`:
  ```agda
  module M (_ : Set₁) where

  record R : Set₁ where
    field
      A : Set
  ```
* `Bug.agda`:
  ```agda
  open import M Set

  r : R
  r = ?
  ```
* `script`:
  ```
  #!/bin/sh

  AGDA=$1

  $AGDA --interaction --caching --no-libraries --ignore-interfaces <<EOF
  IOTCM "Bug.agda" None Indirect (Cmd_load "Bug.agda" [])
  IOTCM "Bug.agda" None Indirect (Cmd_make_case 0 noRange "")
  IOTCM "Bug.agda" None Indirect (Cmd_load "Bug.agda" [])
  IOTCM "Bug.agda" None Indirect (Cmd_make_case 0 noRange "")
  EOF
  ```

Result of running `./script agda`:
```
Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Bug (/home/nad/research/dtp/agda-bug/Bug.agda).\n" t)
(agda2-info-action "*Type-checking*" " Checking M (/home/nad/research/dtp/agda-bug/M.agda).\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : R\n" nil)
((last . 1) . (agda2-goals-action '(0)))
Agda2> ((last . 2) . (agda2-make-case-action '(".M.R.A r = ?")))
((last . 1) . (agda2-goals-action '(0)))
Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Bug (/home/nad/research/dtp/agda-bug/Bug.agda).\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : R\n" nil)
((last . 1) . (agda2-goals-action '(0)))
Agda2> (agda2-info-action "*Error*" "/home/nad/research/dtp/agda-bug/Bug.agda:4,5-6\nCannot split here, as we are not in a function definition\nwhen checking that the expression ? has type R" nil)
((last . 3) . (agda2-maybe-goto '("/home/nad/research/dtp/agda-bug/Bug.agda" . 30)))
(agda2-highlight-load-and-delete-action "/tmp/agda2-mode30495-0")
(agda2-status-action "")
Agda2> 
```

This is a regression, Agda 2.5.1.1 does not have this problem. (However, note that Agda 2.5.1.1 does not support the `--no-libraries` flag used in `script`.)

**Comment by nad** [comment]
I used bisection to search for the first commit that prints `Cannot split here` for the code above, and that seems to be the commit that introduced this error message (446674e487b4e59440f07678e38c03d87e957f1b): "Fixed #289 (see #2006) no more shutter with disgusting findClause". The previous commit on the stable branch (a06c2095c7668a34406c4c7f4c8f4be1ee17af86) seems to work fine (in this particular case), so it seems as if @andreasabel introduced this regression.

**Comment by andreasabel** [comment]
Maybe the `registerInteractionPoint` / `connectInteractionPoint` logic does not work together with caching.
